### DISCUSSION TITLE
ONNX Build Issue

### DISCUSSION THREAD

**Discussion by BMeetu** [onnx/onnx#5012] [Q&A]
Hi,

I am facing below issue while building onyx
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libprotobuf.a(arena.o): relocation R_X86_64_TPOFF32 against symbol `_ZN6google8protobuf8internal9ArenaImpl13thread_cache_E' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libprotobuf.a(common.o): relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC

Please Guide

--M

**Comment by jcwchen** [comment]
Hi @BMeetu,
Similar to https://github.com/onnx/onnx/issues/3570. Have you set `export CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"` before building onnx? If no, please `git clean -xdf` and `export CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"`  and then build onnx again.

**Comment by BMeetu** [comment]
Hi jcwchen,

Thank you for your input. I tried this but still ending up in the same error.

Anything else you can suggest.

--M

**Reply by jcwchen** [reply]
Are you using protoc from apt-get? If yes, as https://github.com/onnx/onnx/issues/3570#issuecomment-885757750 mentioned, I would suggest you remove existing protoc and then try ` apt-get install libprotobuf-dev protobuf-compiler`, `git clean -xdf` onnx build files, and then build onnx again.
